Core/src/org/sleuthkit/autopsy/geolocation/datamodel/GeoLocationParseResult.java

```diff
@@ -0,0 +1,166 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.geolocation.datamodel;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.collections4.ListUtils;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+
+/**
+ * The result of attempting to parse a GeoLocation object.
+ */
+public class GeoLocationParseResult<T> {
+
+    private final boolean successfullyParsed;
+    private final GeoLocationDataException exception;
+    private final BlackboardArtifact artifact;
+    private final T geoObject;
+
+    GeoLocationParseResult(BlackboardArtifact artifact,
+            boolean successfullyParsed, GeoLocationDataException exception, T geoObject) {
+
+        this.successfullyParsed = successfullyParsed;
+        this.exception = exception;
+        this.artifact = artifact;
+        this.geoObject = geoObject;
+    }
+
+    /**
+     * Creates a GeoLocationParseResult as the result of a successful parse.
+     * @param <T>           The type of object to parse.
+     * @param artifact      The artifact that was parsed or failed in parsing
+     *                      (could be null).
+     * @param geoObject     The GeoLocation object to include in result.
+     *
+     * @return The generated WaypointParseResult.
+     */
+    public static <T> GeoLocationParseResult<T> create(BlackboardArtifact artifact, T geoObject) {
+        if (geoObject == null) {
+            return new GeoLocationParseResult(artifact, false, new GeoLocationDataException("GeoLocation object provided was null"), null);
+        }
+
+        return new GeoLocationParseResult(artifact, true, null, geoObject);
+    }
+
+    /**
+     * Creates a GeoLocationParseResult indicating a failed parsing.
+     *
+     * @param <T>           The type of GeoLocation object that was supposed to
+     *                      be parsed.
+     * @param artifact      The artifact that was parsed or failed in parsing
+     *                      (could be null).
+     * @param exception     The exception generated.
+     *
+     * @return The GeoLocationParseResult indicating an error.
+     */
+    public static <T> GeoLocationParseResult<T> error(BlackboardArtifact artifact, GeoLocationDataException exception) {
+        return new GeoLocationParseResult(artifact, false, exception, null);
+    }
+
+    /**
+     * The result of splitting an iterable of GeoLocationParseResults into a
+     * list of successfully parsed items and a list of failures.
+     *
+     * @param <T> The parsed item type.
+     */
+    public static class SeparationResult<T> {
```

Comment:
What is the need for separating the failed vs the parsed? From what I can tell, you just use the returned lists to answer boolean queries (not null, is empty, etc).

Reply:
I initially created this because of a misunderstanding I had that I would need to report the track items that failed to parse to the user along with returning the items that successfully parsed.  Since the error that is reported to the user is generalized, this could potentially be simplified, but methods like `GeoPath.getTracks` and 'WaypointBuilder.getWaypointForArtifact` will need to have something that returns an object containing the list of successfully parsed items for display and some indication that there was an error.